### ISSUE TITLE
LoggingApplicationListener adds default logging categories even if logging.config is set

### ISSUE THREAD

**Issue by avallen** [spring-projects/spring-boot#5375]
This is contrary to the expectation caused by the class javadoc:

> An ApplicationListener that configures the LoggingSystem. If the environment contains a logging.config property a then that will be used to initialize the logging system, otherwise a default configuration is used.

**Comment by snicoll** [comment]
What makes you think it's contrary to the Javadoc exactly? This is about the appender and the format of the log message. If you don't specify any file, then we will provide a default log format for instance. But we will apply the logging levels regardless.

Perhaps you could say what's confusing in that Javadoc so that we update it?

**Comment by avallen** [comment]
the usual logging configuration does not only contain appender and format configurations but also the configuration of the log levels, which is an essential part of how it is used to "initialize the logging system".

My concern is however not only the javadoc that misled me, but the fact that logging categories are set at all by framework code. And especially the spring-foreign org.hibernate.SQL logging category which is _very_ verbose.

I remember cluelessly searching time and again over the previous months through logback.xml configuration files and also through hibernate's configuration searching for the setting that activated the very verbose org.hibernate.SQL logging which to all appearance should not have been activated.

Finally it took me a breakpoint in the logback library to find the culprit in spring boot.

**Comment by snicoll** [comment]
And where that culprit would be? I am sorry but I am not aware of anything that set `org.hibernate.SQL` to `debug`

**Comment by avallen** [comment]
It is one of the logging categories set by 
`org.springframework.boot.logging.LoggingApplicationListener` if the environment contains the debug property.

**Comment by snicoll** [comment]
I will have a look at the Javadoc but everything looks legit from my side. Check [the doc](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-logging-console-output)

> When the debug mode is enabled, a selection of core loggers (embedded container, Hibernate and Spring) are configured to output more information. Enabling the debug mode does not configure your application log all messages with DEBUG level.

Reading the doc is probably faster than adding a breakpoint in the code :)

**Comment by avallen** [comment]
Looking at all places where in the documentation the --debug mode is described I see a problem in that it is easily used by people like me who are unaware of its double function.

Its main use case - from number of mentions in the docs and in when searching for it via google - is the  enabling the logging of the autoconfiguration report - only once at the location you point out is it described as manipulating the logging configuration. So that already sets it up for the kind of oversight that I suffered from.

I can see a few countermeasures for saving people like me from this oversight:

- mention the sideffect of setting logging categories to debug wherever the doc suggests use of the debug flag
- log the reason for the activation of the additional logging categories. I assume this might not be possible during the initialization, maybe it can instead be postponed similar to how the auto-configuration report is logged. Until now the only logging output is generated by logback's debug mode, but it does not say _why_ it activates the logging categories.

Compounding this for me was #5374 which makes the `LoggingApplicationListener` mis-interpret (ignore)  `debug=false`.

Also once at the documentation, you may also document that the LoggingApplicationListener also evaluates a "trace" environment property and enables additional logging categories at trace level - this is as of yet undocumented.
